generator/vulkan/render.zig

```diff
@@ -930,34 +933,54 @@ fn Renderer(comptime WriterType: type) type {
             try self.writer.writeAll("};\n");
         }
 
-        fn renderWrappers(self: *Self) !void {
-            try self.renderWrappersOfDispatchType("BaseWrapper", .base);
-            try self.renderWrappersOfDispatchType("InstanceWrapper", .instance);
-            try self.renderWrappersOfDispatchType("DeviceWrapper", .device);
+        fn renderDispatchTables(self: *Self) !void {
+            try self.renderDispatchTablesOfDispatchType("BaseDispatch", .base);
+            try self.renderDispatchTablesOfDispatchType("InstanceDispatch", .instance);
+            try self.renderDispatchTablesOfDispatchType("DeviceDispatch", .device);
         }
 
-        fn renderWrappersOfDispatchType(self: *Self, name: []const u8, dispatch_type: CommandDispatchType) !void {
+        fn renderDispatchTablesOfDispatchType(self: *Self, name: []const u8, dispatch_type: CommandDispatchType) !void {
             try self.writer.print(
-                \\pub fn {s}(comptime Self: type) type {{
+                \\pub fn {s}(comptime cmds: anytype) type {{
+                \\    comptime var fields: [cmds.len]std.builtin.TypeInfo.StructField = undefined;
+                \\    inline for (cmds) |cmd, i| {{
+                \\        const cmd_type_name = "Pfn" ++ cmd;
+                \\        const cmd_type = @field(GlobalScope, cmd_type_name);
+                \\        fields[i] = .{{
+                \\            .name = "vk" ++ cmd,
+                \\            .field_type = cmd_type,
+                \\            .default_value = @as(?cmd_type, null),
+                \\            .is_comptime = false,
+                \\            .alignment = @alignOf(*cmd_type),
+                \\        }};
+                \\    }}
+                \\    const Dispatch = @Type(.{{ .Struct = .{{
+                \\        .layout = .Auto,
+                \\        .fields = &fields,
+                \\        .decls = &[_]std.builtin.TypeInfo.Declaration{{}},
+                \\        .is_tuple = false,
+                \\    }}}});
                 \\    return struct {{
-                \\
+                \\        dispatch: Dispatch,
```

Comment:
This doesn't have to be a nested struct once the new function syntax is out.

Reply:
Should we perhaps also export Dispatch? I'm not sure in what situation that would ever be useful though, but it would entail putting `pub const Dispatch` in the struct instead of the function body.

Reply:
Personally I don't see the use case for this